### DISCUSSION TITLE
Why you guys do not use python buildin dataclass type?

### DISCUSSION THREAD

**Discussion by yingshaoxo** [fastapi/fastapi#9235] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import datetime

from pydantic import BaseModel


class User(BaseModel):
    id: int
    name = "John Doe"
    signup_ts: datetime | None = None
    friends: list[int] = []


external_data = {
    "id": "123",
    "signup_ts": "2017-06-01 12:22",
    "friends": [1, "2", b"3"],
}
user = User(**external_data)
print(user)
# > User id=123 name='John Doe' signup_ts=datetime.datetime(2017, 6, 1, 12, 22) friends=[1, 2, 3]
print(user.id)
# > 123
```

### Description

For example, in here, I think python's dataclass could do the same thing:

https://fastapi.tiangolo.com/python-types/#pydantic-models

### Operating System

Other

### FastAPI Version

all version

### Python Version

all version

**Comment by yinziyan1206** [comment, marked as answer]
Here is an example
```Python

class AModel(BaseModel):
    id: int
    name: str
    child: "AModel"


@dataclasses.dataclass
class BModel:
    id: int
    name: str
    child: "BModel"


app = FastAPI()


@app.post("/test/a")
async def get_a(a: AModel):
    return a.dict()


@app.post("/test/b")
async def get_b(b: BModel):
    return dataclasses.asdict(b)
```
You can check it if it is wrong in `get_b`

**Comment by yingshaoxo** [comment]
Alright, that's work. Thanks.

___

But here's another problem, how can I ignore a specific type in fastapi
class verification?

For example, I have an enum type inside of a dataclass:

```python
class UserStatus(Enum):
    OFFLINE = 0
    ONLINE = 1

@dataclasses.dataclass
class BModel:
    id: int
    name: str
    status: UserStatus

```

When I transfer it, i'd like to convert the `status` into a string
`OFFLINE` or `ONLINE` than index numbers.

If I give that data to fastapi directly, it will raise an error:

```
  File "/opt/homebrew/lib/python3.10/site-packages/fastapi/routing.py",
line 233, in app
    raise RequestValidationError(errors, body=body)
fastapi.exceptions.RequestValidationError: 1 validation error for Request
body -> user_status
  value is not a valid enumeration member; permitted: 0, 1
(type=type_error.enum; enum_values=[<UserStatus.OFFLINE: 0>,
<UserStatus.ONLINE: 1>])
```

How can I let the fastapi totally ignore the `Enum` type?
